src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -463,37 +501,31 @@ internal static RuntimeAssembly InternalLoad(AssemblyName assemblyRef, ref Stack
             return assembly;
         }
 
-        // FIXME: Merge some of these
-
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        private extern string get_location();
-
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        private static extern string? get_code_base(Assembly a);
-
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        private static extern string get_fullname(Assembly a);
+        private static extern void GetInfo(QCallAssembly assembly, ObjectHandleOnStack res, AssemblyInfoKind kind);
 
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        private static extern string InternalImageRuntimeVersion(Assembly a);
-
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        private extern bool GetManifestResourceInfoInternal(string name, ManifestResourceInfo info);
+        private string GetInfo(AssemblyInfoKind kind)
```

Comment:
What do you think about returning `string?` here and adding `!` in the callers where we know it's going to be non-null?   It feels wrong to say we never return NULL here when it's actually pretty common for CodeBase to be null on wasm, for example.